tests/cnf/core/network/internal/netenv/netenv.go

```diff
@@ -234,3 +238,109 @@ func RemoveAllPoliciesAndWaitForSriovAndMCPStable() error {
 		netinittools.APIClient, netparam.MCOWaitTimeout, time.Minute,
 		netinittools.NetConfig.CnfMcpLabel, netinittools.NetConfig.SriovOperatorNamespace)
 }
+
+// BuildRoutesMapWithSpecificRoutes creates a route map with specific routes.
+func BuildRoutesMapWithSpecificRoutes(podList []*pod.Builder, workerNodeList []*nodes.Builder,
+	nextHopList []string) (map[string]string, error) {
+	if len(podList) == 0 {
+		glog.V(90).Infof("Pod list is empty")
+
+		return nil, fmt.Errorf("pod list is empty")
+	}
+
+	if len(nextHopList) == 0 {
+		glog.V(90).Infof("Nexthop IP addresses list is empty")
+
+		return nil, fmt.Errorf("nexthop IP addresses list is empty")
+	}
+
+	if len(nextHopList) < len(podList) {
+		glog.V(90).Infof("Number of speaker IP addresses[%d] is less than the number of pods[%d]",
+			len(nextHopList), len(podList))
+
+		return nil, fmt.Errorf("insufficient speaker IP addresses: got %d, need at least %d",
+			len(nextHopList), len(podList))
+	}
+
+	routesMap := make(map[string]string)
+
+	for _, frrPod := range podList {
+		if frrPod.Definition.Spec.NodeName == workerNodeList[0].Definition.Name {
+			routesMap[frrPod.Definition.Spec.NodeName] = nextHopList[1]
+		} else {
+			routesMap[frrPod.Definition.Spec.NodeName] = nextHopList[0]
+		}
+	}
+
+	return routesMap, nil
+}
+
+// SetStaticRoute could set or delete static route on all Speaker pods.
+func SetStaticRoute(frrPod *pod.Builder, action, destIP, containerName string,
+	nextHopMap map[string]string) (string, error) {
+	buffer, err := frrPod.ExecCommand(
+		[]string{"ip", "route", action, destIP, "via", nextHopMap[frrPod.Definition.Spec.NodeName]}, containerName)
+	if err != nil {
+		if strings.Contains(buffer.String(), "File exists") {
+			glog.V(90).Infof("Warning: Route to %s already exist", destIP)
+
+			return buffer.String(), nil
+		}
+
+		if strings.Contains(buffer.String(), "No such process") {
+			glog.V(90).Infof("Warning: Route to %s already absent", destIP)
+
+			return buffer.String(), nil
+		}
+
+		return buffer.String(), err
+	}
+
+	return buffer.String(), nil
+}
+
+// CreateExternalNad creats an external network-attchment-definition using the br-ex interface.
+func CreateExternalNad(apiClient *clients.Settings, name, testNameSpace string) error {
```

Comment:
I think, we shoould move this func to the nad.go. WDYT?

Reply:
moved to nad